spec/rubocop/cop/rspec/sort_metadata_spec.rb

```diff
@@ -0,0 +1,217 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::SortMetadata do
+  it 'does not register an offense when using only symbol metadata ' \
+     'in alphabetical order' do
+    expect_no_offenses(<<~RUBY)
+      RSpec.describe 'Something', :a, :b do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using only symbol metadata, ' \
+     'but not in alphabetical order' do
+    expect_offense(<<~RUBY)
+      RSpec.describe 'Something', :b, :a do
+                                  ^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      RSpec.describe 'Something', :a, :b do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when using only a hash of metadata ' \
+     'with keys in alphabetical order' do
+    expect_no_offenses(<<~RUBY)
+      context 'Something', baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using only a hash of metadata, ' \
+     'but with keys not in alphabetical order' do
+    expect_offense(<<~RUBY)
+      context 'Something', foo: 'bar', baz: true do
+                           ^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      context 'Something', baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when using mixed metadata ' \
+     'in alphabetical order (respectively)' do
+    expect_no_offenses(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata, ' \
+     'but only the hash keys are in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :b, :a, baz: true, foo: 'bar' do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      it 'Something', :a, :b, baz: true, foo: 'bar' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense when using mixed metadata, ' \
+     'but only the symbol keys are in alphabetical order' do
+    expect_offense(<<~RUBY)
+      it 'Something', :a, :b, foo: 'bar', baz: true do
+                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Sort metadata alphabetically.
```

Comment:
When seeing this case, I wondered if we should have a separate offense for the symbols and for the hash metadata? In a case like `:y, :x, a: "b"`, would users feel that we ask them to write `a: "b", :x, :y` (which wouldn’t work)?

Maybe I’m overreacting and there is no problem at all.